packages/eds-core-react/src/components/Button/ToggleButton/ToggleButton.tsx

```diff
@@ -0,0 +1,78 @@
+import {
+  HTMLAttributes,
+  cloneElement,
+  Children as ReactChildren,
+  forwardRef,
+  ReactElement,
+  useState,
+  useEffect,
+  isValidElement,
+  AllHTMLAttributes,
+} from 'react'
+import { Button, ButtonProps, ButtonGroupProps } from '..'
+
+export type ToggleButtonProps = {
+  /** Multiple */
+  multiple?: boolean
+  /** Array of selected indexses. */
+  selectedIndexes?: number[]
+  /** OnChange */
+  onChange?: (indexes: number[]) => void
+} & Omit<HTMLAttributes<HTMLElement>, 'onChange'> &
+  Pick<ButtonGroupProps, 'vertical'>
+
+export const ToggleButton = forwardRef<HTMLDivElement, ToggleButtonProps>(
+  function ToggleButton(
+    { children, multiple, selectedIndexes, onChange, ...props },
+    ref,
+  ) {
+    const [pickedIndexes, setPickedIndexes] = useState<number[]>(
+      selectedIndexes || [],
+    )
+
+    useEffect(() => {
+      if (Array.isArray(selectedIndexes)) {
+        setPickedIndexes(selectedIndexes)
+      }
+    }, [selectedIndexes])
+
+    const updatedChildren = ReactChildren.map(
+      children,
+      (child, index: number) => {
+        const isSelected = pickedIndexes.includes(index)
+        const childElement = child as ReactElement<
+          AllHTMLAttributes<HTMLElement>
+        >
+
+        const buttonProps: ButtonProps = {
+          'aria-pressed': isSelected ? true : undefined,
+          variant: isSelected ? 'contained' : 'outlined',
+          onClick: () => {
+            childElement.props?.onClick
+            let updatedSelection = [index]
+
+            if (multiple) {
+              updatedSelection = pickedIndexes.includes(index)
+                ? pickedIndexes.filter((i) => i !== index)
+                : [...pickedIndexes, index]
+            }
+            setPickedIndexes(updatedSelection)
+            if (onChange) {
+              onChange(updatedSelection)
+            }
+          },
+        }
```

Comment:
All of this is related to overriding the child props with `cloneElement` which we only do if the child is a valid element. Suggest moving it inside the if statement as there is no need for this code to run if we aren't calling cloneElement?